### DISCUSSION TITLE
Location of images that Prawn can read

### DISCUSSION THREAD

**Discussion by legacy370** [prawnpdf/prawn#1305] [Q&A]
The application renders images , as long as they are in app/assets/images.  To provide users with more flexibility, it would be nice if Prawn could render images located elsewhere, outside the Rails application.
Is that possible?  If so, what would be the syntax?

**Comment by gettalong** [comment]
It doesn't really matter to Prawn from where you get your image. You can either provide a path to a file or an IO object. The following invocations are equal:

~~~ ruby
Prawn::Document.generate('/tmp/p.pdf') { image('example.jpg') }
Prawn::Document.generate('/tmp/p1.pdf') { image(File.open('example.jpg')) }
~~~

So if the images are outside the Rails application folder on the file system, just specify the correct path. Or if they are not available on the file system, provide an IO object.
